sphinx/ext/napoleon/docstring.py

```diff
@@ -773,6 +783,106 @@ def _strip_empty(self, lines: List[str]) -> List[str]:
         return lines
 
 
+def _recombine_set_tokens(tokens):
+    def takewhile_set(iterable):
+        yield "{"
+
+        open_braces = 1
+        while True:
+            try:
+                token = next(iterable)
+            except StopIteration:
+                if open_braces != 0:
+                    location = ("", "")
+                    logger.warning(
+                        __("invalid value set: %r"),
+                        "".join(tokens),
+                        location=location,
```

Comment:
this warning is not as useful as it could be since there's no information on the location of the invalid line. Is there a way to get that from the attributes of `NumpyDocstring`?

Reply:
also, my test project suggests that this code is run twice and thus two identical warnings are raised. Is that intentional or a bug, or is this a problem with my configuration?